pkg/diagnostics/grpc_tracing.go

```diff
@@ -26,41 +26,35 @@ const grpcTraceContextKey = "grpc-trace-bin"
 // SetTracingInGRPCMiddlewareUnary sets the trace context or starts the trace client span based on request
 func SetTracingInGRPCMiddlewareUnary(appID string, spec config.TracingSpec) grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-		sc := GetSpanContextFromGRPC(ctx, spec)
-		newCtx := NewContext(ctx, sc)
-
-		var err error
-		var resp interface{}
-
-		// 1. check if tracing is enabled or not
-		// 2. if tracing is disabled, set the trace context and call the handler
-		// 3. if tracing is enabled, start the client or server spans based on the request and call the handler with appropriate span context
-		if !diag_utils.IsTracingEnabled(spec.SamplingRate) {
-			resp, err = handler(newCtx, req)
-			return resp, err
-		}
-
 		var span *trace.Span
 		spanName := info.FullMethod
 
+		sc, _ := SpanContextFromGRPCMetadata(ctx)
+		sampler := diag_utils.TraceSampler(spec.SamplingRate)
+
+		var spanKind trace.StartOption
+
+		// This middleware is shared by internal gRPC for service invocation and api
+		// so that it needs to handle separately.
 		if isInternalCalls(info.FullMethod) {
-			_, span = StartTracingServerSpanFromGRPCContext(newCtx, spanName, spec)
+			// For dapr.proto.internals package, this generates ServerSpan.
+			spanKind = trace.WithSpanKind(trace.SpanKindServer)
 		} else {
-			// Instead of generating the span in direct_messaging, dapr changes the spanname
-			// to CallLocal.
+			// For dapr.proto.runtime package, this generates ClientSpan because each Dapr service API
+			// is treat as dependency or internal calls.
 			if spanName == "/dapr.proto.runtime.v1.Dapr/InvokeService" {
```

Comment:
We're hard coding the package version `v1` here. If this changes, the runtime will break.

Reply:
I can fix it